thinc/backends/_custom_kernels.py

```diff
@@ -273,64 +299,80 @@ def backprop_gelu(
     return out
 
 
-def backprop_maxout(dY, which, P, out=None, threads_per_block=128, num_blocks=128):
+def backprop_maxout(dY, which, P, threads_per_block=128, num_blocks=128):
+    _check_array(dY)
+
     B = dY.shape[0]
     I = dY.shape[1]
-    if out is None:
-        out = cupy.zeros((B, I, P), dtype="f")
+
+    out = cupy.zeros((B, I, P), dtype="f")
+
+    _check_which_maxout(which, B, I, P)
+
     backprop_maxout_kernel(
         (num_blocks,), (threads_per_block,), (out, dY, which, B, I, P)
     )
     return out
 
 
-def backprop_mish(dY, X, out=None, threshold=5, threads_per_block=128, num_blocks=128):
-    if out is None:
-        out = cupy.empty_like(X)
+def backprop_mish(
+    dY, X, inplace: bool = False, threshold=5, threads_per_block=128, num_blocks=128
+):
+    _check_array(dY)
+    _check_array(X, dY.shape)
```

Comment:
To me it might be clearer to have something like `_check_compatible_shapes(X, Y)` or if its too long then maybe `_check_compatible(X, Y)`, when the question is whether the shape of `X` and `Y` are the same.

Reply:
Renamed to `_check_compatible`. `_check_compatible_shapes` is too specific, since you can also leave out the shape and it will just check the `dtype`.